crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -210,11 +214,18 @@ impl WorkerPool {
                 //
                 // Note: Rng::gen rolls between [0, 1.0) for f32, so this works as expected.
                 let failed = rand::thread_rng().gen::<f32>() < self.failure_rate;
+                let store = self.store.clone();
```

Comment:
Cloning the store is pretty light-weight, right?

Reply:
According to [this](https://docs.rs/tonic/latest/tonic/client/index.html#:~:text=The%20answer%20is%20simply%20to,the%20same%20channel%20for%20communication.) yes:

> Upon using the your generated client, you will discover all the functions corresponding to your rpc methods take &mut self, making concurrent usage of the client difficult. **The answer is simply to clone the client, which is cheap** as all client instances will share the same channel for communication.